dbms/src/Flash/Pipeline/Schedule/Tasks/NotifyFuture.h

```diff
@@ -24,11 +24,10 @@ struct NotifyFuture
     virtual ~NotifyFuture() = default;
     virtual void registerTask(TaskPtr && task) = 0;
 };
-using NotifyFuturePtr = std::shared_ptr<NotifyFuture>;
 
-extern thread_local NotifyFuturePtr current_notify_future;
+extern thread_local NotifyFuture * current_notify_future;
```

Comment:
Looks like this change may increase the risk of using a object that have been already released, is it by design that the `current_notify_future` will not be released during its lifecycle?

Reply:
yes, `current_notify_future` is only released when the pipeline task changes from `wait-for-notify` to `running`